### PR TITLE
Improvements to CISA

### PR DESCRIPTION
**Pull request by tdcthosc**
# Description

* Test-MtCisaDkim add skip logic for onmicrosoft.com domains that you can't configure DKIM on
* Test-MtCisaAttachmentFilter remove large comment as we have git history if needed
* Test-MtCisaAuditLogPremium remove large comment as we have git history if needed
* CISA.MS.EXO.17.2 do not alter test names but use tag instead if deprecated
* Test-MtCisaBlockHighRiskUsers.Tests.ps1 formatting
* Test-MtCisaAssignmentNotification list misconfigured roles in
* Test-MtCisaAuthenticatorContext.md add direct link in  to Entra
* Test-MtCisaDiagnosticSettings add logic to skip if authorized for scope
* Test-MtCisaPermanentRoleAssignment.md spelling
* Test-MtCisaPermanentRoleAssignment move %TestResult%
* Test-MtCisaPermanentRoleAssignment list misconfigured roles
* ConvertFrom-MailAuthenticationRecordDkim use default DNS if not specified
* ConvertFrom-MailAuthenticationRecordDkim return if no record is found
* ConvertFrom-MailAuthenticationRecordDmarc use default DNS if not specified
* ConvertFrom-MailAuthenticationRecordMx use default DNS if not specified
* ConvertFrom-MailAuthenticationRecordSpf use default DNS if not specified
* Test-MtCisaDmarcRecordExist only process unique domains
* Test-MtCisaDmarcRecordExist skip onmicrosoft.com domains
* Test-MtCisaDmarcRecordReject only process unique domains
* Test-MtCisaDmarcRecordReject skip Microsoft domains
* Test-MtCisaExoAlert markdown update
* Test-MtCisaExoAlert.ps1 minor formatting
* Test-MtCisaExoAlertSiem markdown update
* Test-MtCisaExoAlertSiem.ps1 minor formatting
* Test-MtCisaMalwareZap remove large comment as we have Github history
* Test-MtCisaSpfDirective get all sub-includes if child contains required includes

## Contribution Checklist

Before submitting this PR, please confirm you have completed the following:

- [x] 📖 Read the [guidelines for contributing](https://maester.dev/docs/contributing) to this repository.
- [x] 🧪 Ensure the build and unit tests pass by running `/powershell/tests/pester.ps1` on your local system.

&nbsp;

Join us at the Maester repository [discussions](https://github.com/maester365/maester/discussions) 💬 or [Entra Discord](https://discord.maester.dev/) 🧑‍💻 for more help and conversations!

### CODE REVIEW

File: tests/cisa/exchange/Test-MtCisaAuditLogPremium.Tests.ps1
Comment (SamErde) [inline tests/cisa/exchange/Test-MtCisaAuditLogPremium.Tests.ps1:1]: I like the idea of creating a "Deprecated" tag!

File: powershell/public/cisa/entra/Test-MtCisaDiagnosticSettings.ps1 [hunk 1]
```diff
@@ -54,6 +54,10 @@ function Test-MtCisaDiagnosticSettings {
     $configs = @()
 
     $settings = Invoke-AzRestMethod -Method GET -Path "/providers/microsoft.aadiam/diagnosticSettings?api-version=2017-04-01-preview"
+    if ($settings.StatusCode -ne '200') {
+        Add-MtTestResultDetail -SkippedBecause Custom -SkippedCustomReason 'Not authorized for: /providers/microsoft.aadiam/diagnosticSettings'
```
Comment (SamErde) [inline powershell/public/cisa/entra/Test-MtCisaDiagnosticSettings.ps1]: The `Add-MtTestResultDetail` function includes a predefined `LimitedPermissions` value in the **SkippedBecause** parameter definition. Would it make sense to use this value or create a new one specific to the "Not authorized" response code?

File: powershell/public/cisa/exchange/ConvertFrom-MailAuthenticationRecordDkim.ps1 [hunk 2]
```diff
@@ -40,7 +40,7 @@ function ConvertFrom-MailAuthenticationRecordDkim {
         [string]$DomainName,
 
         # DNS-server to use for lookup.
-        [ipaddress]$DnsServerIpAddress = "1.1.1.1",
+        [ipaddress]$DnsServerIpAddress,
```
Comment (SamErde) [inline powershell/public/cisa/exchange/ConvertFrom-MailAuthenticationRecordDkim.ps1:43]: Is there any risk that `Resolve-DnsName` may return internal records for environments with a split-brain / split-horizon DNS configuration instead of relevant external DNS records?

Should we discuss the use of a default DNS provider? I do understand that some environments may also have external DNS resolution restricted to only allow certain machines to perform external recursion or may only allow it to certain public providers. How can we accommodate both scenarios and inform the user if something doesn't seem right?

File: powershell/public/cisa/exchange/Test-MtCisaAuditLogPremium.ps1 [hunk 3]
```diff
@@ -20,50 +20,4 @@ function Test-MtCisaAuditLogPremium {
 
     Add-MtTestResultDetail -SkippedBecause Custom -SkippedCustomReason "This test has been deprecated by CISA on March 2025. MS.EXO.17.2v1 was originally included in order to enable auditing of additional user actions not captured under Purview Audit (Standard). In October 2023, Microsoft announced changes to its Purview Audit service that included making audit events in Purview Audit (Premium) available to Purview Audit (Standard) subscribers. Now that the rollout of changes has been completed, Purview (Standard) includes the necessary auditing which is addressed by MS.EXO.17.2v1 See [CISA Gov - GitHub](https://github.com/cisagov/ScubaGear/blob/7cefa12639b4bc36990f8f2849b57ad2fdafec4c/PowerShell/ScubaGear/baselines/removedpolicies.md?plain=1#L56)"
     return $null
-
```
Comment (SamErde) [inline powershell/public/cisa/exchange/Test-MtCisaAuditLogPremium.ps1:23]: Large block of testing removed. Is this just because the test is deprecated and you don't want to needlessly run the queries?

File: powershell/public/cisa/exchange/Test-MtCisaAttachmentFilter.ps1 [hunk 4]
```diff
@@ -63,59 +63,4 @@ function Test-MtCisaAttachmentFilter {
     Add-MtTestResultDetail -Result $testResultMarkdown
 
     return $testResult
-
```
Comment (thomas-s-schmidt) [inline powershell/public/cisa/exchange/Test-MtCisaAttachmentFilter.ps1:66]: All of it was commented out instead of removed. No need to keep it as a comment since we have Github as historical reference.

File: powershell/public/cisa/exchange/Test-MtCisaDmarcRecordExist.ps1 [hunk 5]
```diff
@@ -35,7 +35,7 @@ function Test-MtCisaDmarcRecordExist {
 
     $seen = @{}
     $dmarcRecords = @()
-    foreach($domain in $acceptedDomains){
+    foreach($domain in ($acceptedDomains | Get-Unique -AsString)){
```
Comment (SamErde) [inline powershell/public/cisa/exchange/Test-MtCisaDmarcRecordExist.ps1]: The `Get-Unique` cmdlet "requires" items in the list to be sorted for it to work as expected, and is also case-sensitive. Using `Sort-Object -Unique` may provide the desired outcome, as it sorts and is case-insensitive by default.

File: powershell/public/cisa/exchange/Test-MtCisaAuditLogPremium.ps1 (hunk 3 above)
Comment (thomas-s-schmidt) [reply powershell/public/cisa/exchange/Test-MtCisaAuditLogPremium.ps1:23]: All of it was commented out instead of removed. No need to keep it as a comment since we have Github as historical reference.

With regards to the code being commented out in the first place by @merill if I understand it correct, it no longer makes sense to check it, as it will always(?) implicit be enabled from now on?

File: powershell/public/cisa/exchange/ConvertFrom-MailAuthenticationRecordDkim.ps1 (hunk 2 above)
Comment (thomas-s-schmidt) [reply powershell/public/cisa/exchange/ConvertFrom-MailAuthenticationRecordDkim.ps1:43]: I can reason with that. The reason for this change is that most likely, if an enterprise has an internal DNS server, they will have blocked external DNS requests, which is why I think it should default to the system DNS. In general you wouldn't likely have DKIM records on internal DNS servers IMO, meaning it wouldn't result in split-brain.

File: powershell/public/cisa/exchange/Test-MtCisaDmarcRecordExist.ps1 (hunk 5 above)
Comment (thomas-s-schmidt) [reply powershell/public/cisa/exchange/Test-MtCisaDmarcRecordExist.ps1]: That's true, great input 😄  I will adjust.

File: powershell/public/cisa/exchange/ConvertFrom-MailAuthenticationRecordDkim.ps1 (hunk 2 above)
Comment (SamErde) [reply powershell/public/cisa/exchange/ConvertFrom-MailAuthenticationRecordDkim.ps1:43]: I'm good with that. I have seen environments with poorly configured split-brain and things like SPF and DKIM records added on internal zones, but we can't/shouldn't necessarily try to solve for every possible edge case.

File: powershell/public/cisa/entra/Test-MtCisaDiagnosticSettings.ps1 (hunk 1 above)
Comment (merill) [reply powershell/public/cisa/entra/Test-MtCisaDiagnosticSettings.ps1]: Added NotAuthorized

**Review by merill — APPROVED** [review]
LGTM
